### ISSUE TITLE
XGBoost 1.5.0 on Conda: compat.py with FutureWarning

### ISSUE THREAD

**Issue by goncaloperes** [dmlc/xgboost#8448]
Using XGBoost 1.5.0 on Anaconda, on the vevn `Test`, when one imports XGBRegressor as follows

```
from xgboost import XGBRegressor
```

one get the following `FutureWarning`:

```
C:\Users\johndoe\anaconda3\envs\Test\lib\site-packages\xgboost\compat.py:36: FutureWarning:

pandas.Int64Index is deprecated and will be removed from pandas in a future version. Use pandas.Index with the appropriate dtype instead.
```

At this moment it works fine, but in the future that might lead to issues in the future.

I haven't tested importing other methods, but wouldn't be surprised if that is happening with the same.

**Comment by goncaloperes** [comment]
Duplicate of this: https://github.com/dmlc/xgboost/issues/7593

**Comment by hcho3** [comment]
@goncaloperes The issue has been fixed in XGBoost 1.6.0. Please upgrade to the latest XGBoost.

**Comment by goncaloperes** [comment]
@hcho3 I've noticed from #7593. Thank you.
